plugins/catalog-backend/src/service/router.ts

```diff
@@ -132,10 +134,13 @@ function translateQueryToEntityFilters(
       throw new InputError('Complex query parameters are not supported');
     }
 
-    filters.push({
-      key,
-      values: values.map(v => v || null) as string[],
-    });
+    // This one always emits arrays
+    let matchers = filters[key] as (string | null)[];
+    if (!matchers) {
```

Comment:
probably better to `key in filters` here, or `hasOwnProperty`. Could be that you're able to get rid of the cast with that

Reply:
yarp, had to throw in an `a = (b = c)` in there which some may argue is unclean but it sure is prettier